src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -1556,23 +1546,34 @@ void ApplicationManagerImpl::ProcessServiceStatusUpdate(
     const uint32_t connection_key,
     hmi_apis::Common_ServiceType::eType service_type,
     hmi_apis::Common_ServiceEvent::eType service_event,
-    hmi_apis::Common_ServiceUpdateReason::eType service_update_reason) {
+    utils::Optional<hmi_apis::Common_ServiceUpdateReason::eType>
+        service_update_reason) {
   LOG4CXX_AUTO_TRACE(logger_);
 
   LOG4CXX_DEBUG(logger_,
                 "Processing status update with connection key: "
                     << connection_key << " service type: " << service_type
                     << " service_event " << service_event
-                    << "service_update_reason " << service_update_reason);
+                    << " service_update_reason " << service_update_reason);
 
-  const uint32_t app_id = ShouldAddAppIDForService(service_type, service_event)
-                              ? connection_key
-                              : 0;
+  auto app = application(connection_key);
+  auto notification_builder =
+      MessageHelper::ServiceStatusUpdateNotificationBuilder::
+          CreateServiceStatusUpdateNotificationBuilder(service_type,
+                                                       service_event);
 
-  auto notification = MessageHelper::CreateOnServiceStatusUpdateNotification(
-      app_id, service_type, service_event, service_update_reason);
+  const bool should_add_app_id =
+      ShouldAddAppIDForService(service_type, service_event);
```

Comment:
@mked-luxoft we still need this method?

Reply:
@LuxoftAKutsan as i understand, yes, since we still need to check whether it is the first  rpc service for application